stories/__tests__/SQFormCheckboxGroup.stories.test.js

```diff
@@ -0,0 +1,194 @@
+} = composeStories(stories);
+
+const SHOPPING_LIST_OPTIONS = [
+  {label: 'Voltfruit', value: 'voltfruit'},
+  {label: 'Sunshroom', value: 'sunshroom'},
+  {label: 'Bokoblin Guts', value: 'bokoblin guts'},
+  {label: 'Lynel Hoof', value: 'lynel hoof'},
+  {label: 'Stealthfin Trout', value: 'stealthfin trout'}
+];
+
+describe('SQFormCheckboxGroup Tests', () => {
+  describe('Checkboxes Only', () => {
+    it('should render a group and a list of checkboxes', () => {
+      render(<SQFormCheckboxGroup size="auto" />);
+
+      const group = screen.getByRole('group', {name: /shopping list/i});
+      expect(group).toBeInTheDocument();
+
+      const checkboxes = screen.getAllByRole('checkbox');
+      expect(checkboxes).toHaveLength(SHOPPING_LIST_OPTIONS.length);
+    });
+
+    it('should display as checked when initial values are set', () => {
+      const initialValues = {
+        shoppingList: [SHOPPING_LIST_OPTIONS[0].value]
+      };
+
+      render(<SQFormCheckboxGroup SQFormProps={{initialValues}} size="auto" />);
+
+      const checkboxes = screen.getAllByRole('checkbox');
+
+      const targetCheckbox = screen.getByLabelText(
+        SHOPPING_LIST_OPTIONS[0].label
+      );
+
+      const filteredCheckboxes = checkboxes.filter(checkbox =>
+        checkbox.hasAttribute('checked')
+      );
+      expect(filteredCheckboxes).toEqual(
+        expect.arrayContaining([targetCheckbox])
+      );
+      expect(filteredCheckboxes).toHaveLength(1);
+    });
+
+    it('should show as checked when a checkbox is clicked', () => {
+      render(<SQFormCheckboxGroup size="auto" />);
+
+      const checkboxLabel = screen.getByLabelText(
+        SHOPPING_LIST_OPTIONS[0].label
+      );
+      expect(checkboxLabel).not.toBeChecked();
+
+      userEvent.click(checkboxLabel);
+      expect(checkboxLabel).toBeChecked();
+    });
+
+    it('should uncheck when checkbox is checked and user clicks', () => {
+      render(<SQFormCheckboxGroup size="auto" />);
+
+      const checkboxLabel = screen.getByLabelText(
+        SHOPPING_LIST_OPTIONS[1].label
+      );
+      expect(checkboxLabel).not.toBeChecked();
+
+      userEvent.click(checkboxLabel);
+      expect(checkboxLabel).toBeChecked();
+
+      userEvent.click(checkboxLabel);
+      expect(checkboxLabel).not.toBeChecked();
+    });
+
+    it('should render as required when isRequired is true', () => {
+      render(<SQFormCheckboxGroup size="auto" isRequired />);
+
+      const required = screen.getByText('Required');
+      expect(required).toBeInTheDocument();
+    });
+
+    it('should render an extra checkbox when adding select all capability', () => {
+      render(<SQFormCheckboxGroup size="auto" shouldUseSelectAll />);
+
+      const selectAllCheckbox = screen.getByLabelText('All');
+      expect(selectAllCheckbox).toBeInTheDocument();
+      expect(selectAllCheckbox).not.toBeChecked();
+    });
+
+    it('should check all checkboxes when select all checkbox checked', () => {
```

Comment:
What do we think about having a test that tests the 'selectall' checkbox in a situation where some of the options (not all) are selected to ensure that the select all isn't simply toggling the checkbox and actually doing what it should.

Reply:
Good idea! I'll add a scenario like that